pkg/sql/schemachanger/scdecomp/decomp.go

```diff
@@ -229,11 +231,36 @@ func (w *walkCtx) walkRelation(tbl catalog.TableDescriptor) {
 	switch {
 	case tbl.IsSequence():
 		w.ev(descriptorStatus(tbl), &scpb.Sequence{
-			SequenceID:  tbl.GetID(),
-			IsTemporary: tbl.IsTemporary(),
+			SequenceID:          tbl.GetID(),
+			IsTemporary:         tbl.IsTemporary(),
+			OptionalRestartWith: nil, /* Only used during creation. */
 		})
 		if opts := tbl.GetSequenceOpts(); opts != nil {
 			w.backRefs.Add(opts.SequenceOwner.OwnerTableID)
+			addSequenceOption := func(key string, value interface{}) {
+				// Nil or empty keys can be skipped.
+				if value == nil {
+					return
+				}
+				keyStr := fmt.Sprintf("%v", value)
+				if len(keyStr) == 0 {
+					return
+				}
```

Comment:
You need to also return when the value is 0 (or whatever the default is) for integer values. Perhaps pass an extra `defaultValue` argument to `addSequenceOption` and compare the value to that?